juju/terraform-provider-juju
# Migrate model resource to framework

## Description

Migrates the model resource from sdk2 to terraform provider framework.

## Type of change

- Maintenance work (repository related, like Github actions, or revving the Go version, etc.)

## Environment

- Juju controller version: 2.9

- Terraform version: 1.5.4

## QA steps

Create an integration resource (see `examples/resources/juju_model`) with `0.7.0`. 

Get this change, and :
```sh
 $ make install
```
-- cd into wherever the plan is --

Change the version in the plan to `0.9.0`:

```terraform
terraform {
  required_providers {
    juju = {
      version = "0.9.0"
      source  = "juju/juju"
    }
  }
}
```

Then run:

```sh
 $ terraform init --upgrade  && terraform plan && terraform apply --auto-approve
```

You should see `No changes ...` output.
